### ISSUE TITLE
SharedBlock 'block.page.id' is null and bypass 'is defined'

### ISSUE THREAD

**Issue by epixerion** [sonata-project/SonataPageBundle#744]
Using a sharedBlock on my website result in this error 

`Impossible to access an attribute ("id") on a null variable in SonataPageBundle:Block:block_base.html.twig at line 23.`

the line concerned is `<!-- start rendering, block.id: {{ block.id }} - page.id: {% if block.page is defined %}{{ block.page.id }}{% else %}no related page{% endif %} - name: {{ block.name }} -->`

apparently, on a shared block, the 'block.page' variable is defined, but set to null, as twig consider a variable defined, even with a null value, its try to display the id of the page

I would recommend to simply change 
`{% if block.page is defined %}{{ block.page.id }}{% else %} `
to 
`{% if block.page is not null %}{{ block.page.id }}{% else %}`

as well as on the line 31.

**Comment by greg0ire** [comment]
Seems like the right fix indeed. Would you like to provide a PR?

**Comment by epixerion** [comment]
I actually never tried to contribute to a github before, I will try at the end of the week as i'm currently in a rush.

I would also like to provide an update for the issues, my fix seem to have broken some default sonata page (the page not found for exemple)
unless I did something wrong before, the fix should be  :
`{% if block.page is defined and block.page is not null  %}{{ block.page.id }}{% else %}no related page{% endif %}`

**Comment by greg0ire** [comment]
> apparently, 'is not null' do no check if the variable is defined first.

That's right, but I couldn't think of a case where block.page wouldn't be defined. I assumed block was a `Block` instance, which always as a `getPage` method. I was wrong, it seems.

> I actually never tried to contribute to a github before

This is a good opportunity if you want to give it a try, because the fix won't be long to do. Please read CONTRIBUTING.md carefully before proceeding. Looking forward to your PR at then end of the week!

**Comment by Aylarius** [comment]
Thanks @epixerion great fix! I had the same issue...

**Comment by greg0ire** [comment]
What about this PR?

**Comment by jayesbe** [comment]
Interesting that this is still an outstanding issue.. I can't even login because of this.
